### ISSUE TITLE
DINO Loss is not decreasing

### ISSUE THREAD

**Issue by mkh1989** [AICONSlab/3DINO#6]
Hi,

Thank you for sharing the repo. 

I am using this repo to pretrain a model with my dataset. Currently, I am using 10k 3D images taken from one organ. The problem that I have is shown in the attached image; DINO loss does not decrease. Here is my config:

dino:
  loss_weight: 1.0
  head_n_prototypes: 100
  head_bottleneck_dim: 256
  head_nlayers: 3
  head_hidden_dim: 2048
  koleo_loss_weight: 0.1
ibot:
  loss_weight: 1.0
  mask_sample_probability: 0.5
  mask_ratio_min_max:
  - 0.2
  - 0.75
  separate_head: true
  head_n_prototypes: 100
  head_bottleneck_dim: 256
  head_nlayers: 3
  head_hidden_dim: 2048
train:
  batch_size_per_gpu: 16

Did you face similar behavior? Do you have any suggestions? 

<img width="1773" height="371" alt="Image" src="https://github.com/user-attachments/assets/1cb96616-a36f-4dd6-9275-3aad3e4dee71" />

**Comment by Tonyxu74** [comment]
Actually, I think this is the expected behavior with DINO-like teacher-student models, since the teacher is a moving target (changes every iteration), the SSL loss does not necessarily monotonically decrease over training. I don't have any examples on hand currently, but this issue has been raised even in the original DINOv2 repo. In this comment the DINOv2 authors provide an example loss graph: https://github.com/facebookresearch/dinov2/issues/143#issuecomment-1651732518, and earlier in the same issue someone posted a loss curve that looks **very** similar to yours. 

To properly monitor pretraining progression, I would set up a downstream task to evaluate on instead (i.e. performing a linear segmentation like we describe in the paper, or doing classification). Thanks for your interest in our work!

**Comment by mkh1989** [comment]
Thanks for the info.

I evaluated the pretrained model for a binary classification task, and the plots show the performance for linear probing (top) and KNN (bottom).

Normally, we expect to see an increase in the performance when the pretraining goes on (more iterations), but figures do not show this. Possibly, there is something wrong during the pretraining phase.

By looking at the dino loss curve, it quickly reaches ln(number of prototypes) and stays there as training continues (ln: natural logarithm). 

I have seen many dino loss curves; almost all of them drop during the early training and reach a plateau. This decrease is not as negligible as observed in my dino loss.

I am wondering if you can share your checklist to tune the hyperparameters of pretraining. As you, in the config file, there are a lot of parameters that must be tuned.  

<img width="1000" height="600" alt="Image" src="https://github.com/user-attachments/assets/4e34407a-b736-494d-a893-206c5dec43ec" />
<img width="1000" height="600" alt="Image" src="https://github.com/user-attachments/assets/46a741ef-a7e1-442d-a75e-e32213e62e55" />

.
